src/CommitmentsV2/SFA.DAS.CommitmentsV2.UnitTests/Application/Commands/ResumeApprenticeshipCommandHandlerTests.cs

```diff
@@ -0,0 +1,177 @@
+                .UseInMemoryDatabase(Guid.NewGuid().ToString())
+                .ConfigureWarnings(w => w.Throw(RelationalEventId.QueryClientEvaluationWarning))
+                .Options);
+
+            _currentDateTime = new Mock<ICurrentDateTime>();
+            _unitOfWorkContext = new UnitOfWorkContext();
+            _handler = new ResumeApprenticeshipCommandHandler(
+                new Lazy<ProviderCommitmentsDbContext>(() => _dbContext),
+                _currentDateTime.Object,
+                _authenticationService.Object,
+                Mock.Of<ILogger<ResumeApprenticeshipCommandHandler>>());
+        }
+
+        [Test]
+        public async Task Handle_WhenHandlingCommand_ResumeApprenticeship_CreatesAddHistoryEvent()
+        {
+            // Arrange
+            var apprenticeship = await SetupApprenticeship();
+
+            var command = new ResumeApprenticeshipCommand
+            {
+                ApprenticeshipId = apprenticeship.Id,
+                UserInfo = new UserInfo()
+            };
+
+            // Act
+            await _handler.Handle(command, new CancellationToken());
+
+            // Simulate Unit of Work contex transaction ending in http request.
+            await _dbContext.SaveChangesAsync();
+
+            // Assert
+            var historyEvent = _unitOfWorkContext.GetEvents().OfType<EntityStateChangedEvent>()
+                .First(e => e.EntityId == apprenticeship.Id);
+            historyEvent.EntityType.Should().Be("Apprenticeship");
+
+            //historyEvent.StateChangeType.Should().Be(UserAction.StopApprenticeship);
+            var definition = new { PaymentStatus = PaymentStatus.Paused };
+            var historyState = JsonConvert.DeserializeAnonymousType(historyEvent.UpdatedState, definition);
+            historyState.PaymentStatus.Should().Be(PaymentStatus.Active);
+        }
+
+        [Test]
+        [InlineAutoData(PaymentStatus.Completed)]
+        [InlineAutoData(PaymentStatus.Withdrawn)]
+        [InlineAutoData(PaymentStatus.Active)]
+        public async Task Handle_WhenHandlingCommand_ThrowErrorForNonPausedApprenticeshipStatus(PaymentStatus paymentStatus)
+        {
+            // Arrange
+            var apprenticeship = await SetupApprenticeship(Party.Employer, paymentStatus);
+
+            var command = new ResumeApprenticeshipCommand
+            {
+                ApprenticeshipId = apprenticeship.Id,
+                UserInfo = new UserInfo()
+            };
+
+            // Act
+            var exception = Assert.ThrowsAsync<DomainException>(async () => await _handler.Handle(command, new CancellationToken()));
+
+            // Assert
+            exception.DomainErrors.Should().BeEquivalentTo(new { ErrorMessage = "Only paused record can be activated" });
+        }
+
+        private ICollection<DataLockStatus> SetupDataLocks(long apprenticeshipId)
```

Comment:
Is datalock setup required for these tests - if not, please remove them as they cause confusion.